src/main/java/com/pubnub/api/workers/SubscribeMessageWorker.java

```diff
@@ -68,6 +68,8 @@ private void takeMessage() {
             } catch (InterruptedException e) {
                 Thread.currentThread().interrupt();
                 log.trace("take message interrupted", e);
+            } catch (Exception e) { // don't crash the thread on malformed messages
+                log.warn("Unexpected message processing error", e);
```

Comment:
Wouldn't decryption error also be `PubNubException` which already caught above (we caught there decryption error)?

Reply:
As I learned from Marcin there are other potential errors that are thrown by parts of the decryption process. While we could try to catch them and convert them into PNException, this safeguard is meant to be wider than _just_ for encryption, that is: if there are other issues with a particular message payload that we weren't aware of, we should ignore the message (since there's nothing we can do with it) and continue processing next messages

Without this catch-all block, the SDK just silently stops processing anything and it's impossible to recover

Reply:
They, should we also add `Thread.currentThread().interrupt();` here?

Reply:
I don't think so, we don't want the thread to exit from the loop. Unless I'm missing something?
(see ` while (!Thread.interrupted()) ` in line 40)

Reply:
I just wanted to point that other unknown error terminate thread and this `all other exceptions` handler won't - a bit of inconsistency.

Reply:
I've remembered about that case. Then maybe @marcin-cebo suggestion is better (I didn't track changes in history, so there is probably a better way).

Reply:
Guys, as we agreed. Let's talk about this case tomorrow on planning.

Reply:
changed according to the conversation yesterdat - decryption errors are  handled by delivering message and setting error flag